OpenNMT/onmt/Models.py

```diff
@@ -99,9 +98,9 @@ def __init__(self, opt, dicts):
 
 
     def forward(self, input, hidden, context, init_output):
-        emb = self.word_lut(input)
+        emb = self.word_lut(input).transpose(0, 1)
```

Comment:
If you're going to be operating as (batch, seq_len, dim) everywhere, maybe it's better to just have the data in this format end-to-end? I.e. starting with the data loader. I'm not sure if this is better, it requires more far-reaching changes.